### DISCUSSION TITLE
The page does not take full height based on its partent?

### DISCUSSION THREAD

**Discussion by maxpaleo** [tailwindlabs/tailwindcss#14071] [Help]
Hi All, 

This is a pretty simple subject, but it seems like I've struggled with it on every project, and have had to find some workaround. 
For some reason, when I use the min-h-full or h-full class on a page div, it does not take the height of its parent. 

**This is the situation:** 

- The root of my app has a min-h-screen class 
- My page (direct child) has a class of min-h-full 

But as you can see in the image below, the page does not take the height of its parent. I can't seem to figure this out, and I always end up using classes like dvh or smh in order to set this, but it seems logically that a page should just take a 100% height of its parent without having to reference screen sizes? 

- The root is shown with the blue border
- The page is shown in red with a red border

<img width="1624" alt="Screenshot 2024-07-26 at 7 29 53 PM" src="https://github.com/user-attachments/assets/d43c1e4b-1803-4c44-b59a-51802944a54f">

**Comment by wongjn** [comment]
As per [the MDN documentation on percentage `min-height` values](https://developer.mozilla.org/en-US/docs/Web/CSS/min-height#formal_definition):

> The percentage is calculated with respect to the height of the generated box's containing block. If the height of the containing block is not specified explicitly (i.e., it depends on content height), and this element is not absolutely positioned, the percentage value is treated as `0`.

The `<body>` element's height depends on content height due to the fact that `min-height` must take into account content height – that's the whole point of the `min-` part. Thus, the `100%` on the inner `<div>` element resolves to `0`.
